sc/memiavl/tree.go

```diff
@@ -195,12 +202,16 @@ func (t *Tree) Has(key []byte) bool {
 }
 
 func (t *Tree) Iterator(start, end []byte, ascending bool) dbm.Iterator {
+	t.mtx.RLock()
```

Comment:
Iterator is tricker, it supposed the hold the RLock before the iterator is closed

Reply:
Hmm yeah, iterator is a bit tricky, this lock only covers the creation of the iterator, might not be enough